### ISSUE TITLE
0.1.0 Release Plan

### ISSUE THREAD

**Issue by jtpio** [jupyterlite/jupyterlite#383]
This is an issue for the `0.1.0` release planning.

We still have the MVP board for the things we wanted to tackle: https://github.com/jupyterlite/jupyterlite/projects/1 (although this was mostly before we have the `jupyterlite` CLI)

Having this in an issue makes it more visible and it can also be pinned.

## Dates

There will be a talk at PyData Global 2021 about JupyterLite: https://pydata.org/global2021/schedule/presentation/113/jupyterlite-jupyter-webassembly-python/

Maybe we can aim for a `0.1.0b0` just before the event, somewhere around Wednesday 27 October. We could then stay in beta for a little while, or just proceed with rc and then final one and two weeks after.

## Issues

- https://github.com/jupyterlite/jupyterlite/pull/371
- https://github.com/jupyterlite/jupyterlite/pull/310
- https://github.com/jupyterlite/jupyterlite/issues/382
- A look at the issues [tagged with bug](https://github.com/jupyterlite/jupyterlite/issues?q=is%3Aissue+is%3Aopen+label%3Abug)

## Nice to have

- Not strictly related to the main `jupyterlite` repo, but it would be great to have Xeus kernels packaged and release:
  - https://github.com/jupyterlite/xeus-kernel for the common package other xeus kernels can depend on
  - https://github.com/DerThorsten/jupyterlite_xeus_lua (to be moved to the `jupyterlite` org later) for a xeus-based Lua kernel
  - https://github.com/DerThorsten/xeus-wren following the same approach for a different language

**Comment by jtpio** [comment]
The first beta is out: https://github.com/jupyterlite/jupyterlite/releases/tag/v0.1.0b0

We can wait for https://github.com/jupyterlite/jupyterlite/issues/511 and make a new beta release with the JupyterLab 3.3 packages (and other beta releases before if needed).

And we can postpone https://github.com/jupyterlite/jupyterlite/issues/386 to `0.2.0`, so there is also time to iterate on the web worker interface (https://github.com/jupyterlite/jupyterlite/issues/463).

**Comment by jtpio** [comment]
@bollwyvl I was thinking about pushing `0.1.0` soon so we have a "proper" release out. Apart from https://github.com/jupyterlite/jupyterlite/pull/525 it's mostly bug fixes and docs remaining, which could go to `0.1.x` as well.

We can then start with `0.2.0` right away with potential new breaking changes such as https://github.com/jupyterlite/jupyterlite/issues/463 and https://github.com/jupyterlite/jupyterlite/issues/386.

**Comment by bollwyvl** [comment]
> soon

welp, seems like lab 3.4.0rc0 and pyodide 0.22.1a0 dropped in the last 24... seems worthy to grab those features.

> potential new breaking changes

I dunno... as we've been reading the issues (graciously well-written, or... otherwise) it seems like people get really bent out of shape about:

- contents in kernels (probably blocked by #463)
- performance (probably blocked by #137)

some things i don't know:

- what kind of time investment we can actually make right now to meet those goals
  - other than how we've been: here and there
- whether we want to lose the _hey, you installed this with `--pre`, you get exactly what you asked for_ retort
  - if folk start making _more_ demands (without contributions) about how this project isn't meeting their exact production needs, i don't know how much fun it's going to be to continue contributing

**Comment by jtpio** [comment]
> welp, seems like lab 3.4.0rc0 and pyodide 0.22.1a0 dropped in the last 24... seems worthy to grab those features.

yes these would be great to get in! And should hopefully be just a matter of bumping dependencies.

> whether we want to lose the _hey, you installed this with `--pre`, you get exactly what you asked for_ retort
> 
>   * if folk start making _more_ demands (without contributions) about how this project isn't meeting their exact production needs, i don't know how much fun it's going to be to continue contributing

Right. Although a `0.1.0` can still be interpreted as very early. Also no problem to leave some kind of disclaimer in the docs or the README to mention things are still moving fast and likely to change in future releases. As always it's an iterative process and folks making demands just have to deal with it, or help by making contributions.

**Comment by kloczek** [comment]
> @bollwyvl I was thinking about pushing `0.1.0` soon so we have a "proper" release out. Apart from #525 it's mostly bug fixes and docs remaining, which could go to `0.1.x` as well.

Is it anything still on outstanding list for final 0.1.0? 🤔

**Comment by kloczek** [comment]
Sorry for asking .. I've done that because `jupyterlite` as it is now other modules started using and lack of official version creates kid of congestion on packaging 😋

**Comment by kloczek** [comment]
May I ask about ET? 🤔  (adking only to be able to coordinate that with other changes)

**Comment by jtpio** [comment]
Hoping for next week or so.

Soon after there will likely be 0.2.0ax pre-release using the JupyterLab 4 and Notebook 7 packages. For JupyterLab 4 and Notebook 7 the goal is to make a final release before JupyterCon 2023 early May. In JupyterLite we consume the lab and notebook packages directly and it would also be nice to have something working with the latest packages before JupyterCon.

**Comment by jtpio** [comment]
`0.1.0` has been released: https://github.com/jupyterlite/jupyterlite/releases/tag/v0.1.0

**Comment by jtpio** [comment]
`0.2.0` is being tracked in https://github.com/jupyterlite/jupyterlite/issues/1038 and in the `0.2.0` milestone: https://github.com/jupyterlite/jupyterlite/milestone/4

**Comment by jtpio** [comment]
> Is there already a plan for `conda-forge`? Happy to do the plumbing.

Yes, started in https://github.com/conda-forge/staged-recipes/pull/22344.
